subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/CrossVersionPerformanceTestRunner.groovy

```diff
@@ -175,42 +172,10 @@ class CrossVersionPerformanceTestRunner extends PerformanceTestSpec {
         perVersion
     }
 
-    private static String resolveVersion(String version, ReleasedVersionDistributions releases) {
-        switch (version) {
-            case 'last':
-                return releases.mostRecentRelease.version.version
-            case 'nightly':
-                return LatestNightlyBuildDeterminer.latestNightlyVersion
-            case 'defaults':
-                throw new IllegalArgumentException("'defaults' shouldn't be used in target versions.")
-            default:
-                def releasedVersion = findRelease(releases, version)
-                if (releasedVersion) {
-                    return releasedVersion.version.version
-                } else if (isRcVersionOrSnapshot(version)) {
-                    // for snapshots, we don't have a cheap way to check if it really exists, so we'll just
-                    // blindly add it to the list and trust the test author
-                    // Only active rc versions are listed in all-released-versions.properties that ReleasedVersionDistributions uses
-                    return version
-                } else {
-                    throw new RuntimeException("Cannot find Gradle release that matches version '$version'")
-                }
-        }
-    }
-
     @VisibleForTesting
     static Iterable<String> toBaselineVersions(ReleasedVersionDistributions releases, List<String> targetVersions, String minimumBaseVersion) {
```

Comment:
I think this whole method should go to `BaselineVersionResolver`. Then it can also be public and the `@VisibleForTesting` can go away.

Reply:
done